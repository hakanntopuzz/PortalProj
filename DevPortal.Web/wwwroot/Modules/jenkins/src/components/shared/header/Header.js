import React, { useState, Fragment } from "react";
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav } from "reactstrap";
import "../../../styles/header.css";
import useFetch from "../../../hooks/useFetch";
import { MENU_URL } from "../../../config/urls";
import initialState from "../../../config/initialState";
import images from "../../../config/images";
import ProfileMenu from "./ProfileMenu";
import NavItems from "./NavItems";

const Header = () => {
  const { data } = useFetch(MENU_URL, initialState.menu);
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  
  const toggle = () => setIsMenuOpen(!isMenuOpen);

  return (
    <header>
      <Navbar color="light" light expand="lg">
        <div className="container-middle d-xl-flex d-lg-flex justify-content-between align-items-center position-relative">
          <NavbarBrand href="/">
            <img src={images.DevPortalLogo} alt="dev portal logo" width="80" />
          </NavbarBrand>
          <NavbarToggler
            onClick={toggle}
            data-testid="toggleMenuButton"
            className="float-right"
          />
          {data && data.length > 0 && (
            <Fragment>
              <ProfileMenu menu={data} />
              <Collapse
                isOpen={isMenuOpen}
                navbar
                data-testid="collapse-menu"
                data-value={isMenuOpen}
                className="justify-content-end">
                <Nav className="nav" navbar data-testid="nav-menu">
                  <NavItems menu={data} />
                </Nav>
              </Collapse>
            </Fragment>
          )}
        </div>
      </Navbar>
    </header>
  );
};

export default Header;
