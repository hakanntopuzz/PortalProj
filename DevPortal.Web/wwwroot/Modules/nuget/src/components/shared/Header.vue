<template>
  <header>
    <nav class="navbar navbar-expand-lg navbar-light bg-light">
      <div class="container-middle d-xl-flex d-lg-flex justify-content-between align-items-center position-relative">
          <a class="navbar-brand" href="/"><img src="/img/devportal-logo.png" alt="dev portal logo" width="80" /></a>
          <div class="nav-item-profile dropdown float-right">
              <a class="nav-link dropdown-toggle" href="#" id="profileDropdown" role="button" data-toggle="dropdown"
                aria-haspopup="true" aria-expanded="false">{{ userIdentityName }}</a>
              <div class="dropdown-menu dropdown-menu-right" aria-labelledby="profileDropdown">
                  <template v-for="item in userProfile">
                    <a v-for="child in item.children" :key="child.name" class="dropdown-item" :href="child.link">
                      <i v-if="child.icon !== null && child.icon !== ''" :class="'fa ' + child.icon" aria-hidden="true"></i>
                      {{ child.name }}
                    </a>
                  </template>
              </div>
          </div>
          <button class="navbar-toggler float-right" type="button" data-toggle="collapse" data-target="#navbarSupportedContent"
                  aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
              <span class="navbar-toggler-icon"></span>
          </button>
          <div class="collapse navbar-collapse justify-content-end" id="navbarSupportedContent">
              <ul class="nav navbar-nav">
                <li class="nav-item dropdown" v-for="item in mainMenu" :key="item.name">
                    <a class="nav-link dropdown-toggle" href="#" id="appsDropdown" role="button" data-toggle="dropdown"
                      aria-haspopup="true" aria-expanded="false">
                        {{ item.name }}
                    </a>
                    <div class="dropdown-menu" aria-labelledby="appsDropdown" v-if="item.children.length > 0">
                      <a class="dropdown-item" :href="child.link" v-for="child in item.children" :key="child.name">{{ child.name }}</a>
                    </div>
                </li>
              </ul>
          </div>
      </div>
    </nav>
  </header>
</template>
<script>
export default {
  data () {
    return {
      mainMenu: [],
      userProfile: [],
      userIdentityName: ''
    }
  },
  created () {
    this.$store.dispatch('getMenu').then(menu => {
      this.mainMenu = menu.filter(x => x.group === 1)
      this.userProfile = menu.filter(x => x.group === 2)
    })

    this.$store.dispatch('getUserIdentityName').then(response => {
      this.userIdentityName = response
    })
  }
}
</script>
<style>
  .dropdown-item {
    padding: 1.25rem 1.5rem;
  }

  .bg-light {
    background-color: #f8f9fa !important;
  }

  .nav-link:hover, .dropdown.show .nav-link, .nav-item.active .nav-link {
    color: #4667dd !important;
  }

  @media (max-width:768px) {
    .dropdown-item {
        padding: .5rem 1rem;
        color: rgba(0,0,0,.5);
    }
  }

  .nav-item-profile {
    color: #fff;
    width: 40px;
    height: 40px;
    background: #f2695c;
    border-radius: 50%;
    text-align: center;
    font-weight: 600;
  }
</style>
