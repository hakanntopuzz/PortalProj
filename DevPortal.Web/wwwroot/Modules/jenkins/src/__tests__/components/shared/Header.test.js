import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import Header from "../../../components/shared/header/Header";
import ShallowRenderer from "react-test-renderer/shallow";
import DevPortalLogo from "../../../img/devportal-logo.png";

describe("Header", () => {
  afterEach(cleanup);

  test("renders without crashing", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Header />);

    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  test("header component has dev portal logo", () => {
    const { getByAltText } = render(<Header />);
    const logo = getByAltText("dev portal logo");

    expect(logo).toBeInTheDocument();
    expect(logo.getAttribute("src")).toBe(DevPortalLogo);
  });

  // test("when clicking toggle menu button set the menu open", () => {
  //   const menu = [
  //     {
  //       name: "Uygulama",
  //       link: null,
  //       group: 1,
  //       children: [
  //         {
  //           name: "Uygulamalar",
  //           link: "/application",
  //           children: [],
  //           hasChildren: false,
  //         },
  //         {
  //           name: "Uygulama Grupları",
  //           link: "/applicationgroup",
  //           children: [],
  //           hasChildren: false,
  //         },
  //       ],
  //       hasChildren: true,
  //     },
  //   ];
    
  //   const mockData = { error: false, loading: false, data: menu };

  //   const { getByTestId } = render(<Header />);

  //   jest.mock("../../../hooks/useFetch", () => {
  //     const fetchData = jest.fn();
  //     return jest
  //       .fn()
  //       .mockImplementation(() => {
  //         return { fetchData: fetchData };
  //       })
  //       .mockReturnValue(Promise.resolve(mockData));
  //   });

  //   const toggleMenuButton = getByTestId("toggleMenuButton");
  //   const collapseMenu = getByTestId("collapse-menu");

  //   fireEvent.click(toggleMenuButton);
  //   expect(collapseMenu.getAttribute("data-value")).toBe("true");
  // });

  // test("header component has a navigation menu", () => {
  //   const menu = [
  //     {
  //       name: "Uygulama",
  //       link: null,
  //       group: 1,
  //       children: [
  //         {
  //           name: "Uygulamalar",
  //           link: "/application",
  //           children: [],
  //           hasChildren: false,
  //         },
  //         {
  //           name: "Uygulama Grupları",
  //           link: "/applicationgroup",
  //           children: [],
  //           hasChildren: false,
  //         },
  //       ],
  //       hasChildren: true,
  //     },
  //   ];
    
  //   const mockData = { error: false, loading: false, data: menu };

  //   const { getByTestId } = render(<Header />);

  //   jest.mock("../../../hooks/useFetch", () => {
  //     const fetchData = jest.fn();
  //     return jest
  //       .fn()
  //       .mockImplementation(() => {
  //         return { fetchData: fetchData };
  //       })
  //       .mockReturnValue(Promise.resolve(mockData));
  //   });

  //   const { getByTestId } = render(<Header />);
  //   const navMenu = getByTestId("nav-menu");

  //   expect(navMenu).toBeInTheDocument();
  // });
});
