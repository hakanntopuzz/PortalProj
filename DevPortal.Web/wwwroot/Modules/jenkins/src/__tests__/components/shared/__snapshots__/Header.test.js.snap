// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<header>
  <Navbar
    color="light"
    expand="lg"
    light={true}
    tag="nav"
  >
    <div
      className="container-middle d-xl-flex d-lg-flex justify-content-between align-items-center position-relative"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        <img
          alt="dev portal logo"
          src="devportal-logo.png"
          width="80"
        />
      </NavbarBrand>
      <NavbarToggler
        className="float-right"
        data-testid="toggleMenuButton"
        onClick={[Function]}
        tag="button"
        type="button"
      />
    </div>
  </Navbar>
</header>
`;
